### ISSUE TITLE
Connection lost

### ISSUE THREAD

**Issue by jialeguan** [iamhyc/Overleaf-Workshop#42]
My old session cookie expired and I tried to use a new cookie looks like this GCLB=[???]; oa=0; overleaf_session2=s%[???].
The project list can be loaded. But when I open any project, the page will just keep loading and notifying connection lost.

**Comment by lucumon** [comment]
Same for me... Maybe overleaf trying to block this as I saw an ad offering copilot for paid plan users?

**Comment by iamhyc** [comment]
The problem is confirmed. It seems like overleaf updates its authentication process when open the project.

I will try to fix it later today. The error log is attached below.

```
mainThreadExtensionService.ts:82 Error: [object Object]
    at SocketNamespace.<anonymous> (C:\Users\Mark Hong\scoop\persist\vscode\data\extensions\iamhyc.overleaf-workshop-0.4.0\out\api\socketio.js:35:19)
    at SocketNamespace.EventEmitter.emit [as $emit] (C:\Users\Mark Hong\scoop\persist\vscode\data\extensions\iamhyc.overleaf-workshop-0.4.0\node_modules\socket.io-client\lib\events.js:165:15)
    at SocketNamespace.onPacket (C:\Users\Mark Hong\scoop\persist\vscode\data\extensions\iamhyc.overleaf-workshop-0.4.0\node_modules\socket.io-client\lib\namespace.js:182:20)
    at Socket.onPacket (C:\Users\Mark Hong\scoop\persist\vscode\data\extensions\iamhyc.overleaf-workshop-0.4.0\node_modules\socket.io-client\lib\socket.js:466:30)
    at WS.Transport.onPacket (C:\Users\Mark Hong\scoop\persist\vscode\data\extensions\iamhyc.overleaf-workshop-0.4.0\node_modules\socket.io-client\lib\transport.js:98:17)
    at WS.Transport.onData (C:\Users\Mark Hong\scoop\persist\vscode\data\extensions\iamhyc.overleaf-workshop-0.4.0\node_modules\socket.io-client\lib\transport.js:69:16)
    at WebSocket.websocket.onmessage (C:\Users\Mark Hong\scoop\persist\vscode\data\extensions\iamhyc.overleaf-workshop-0.4.0\node_modules\socket.io-client\lib\transports\websocket.js:75:12)
    at WebSocket.onMessage (C:\Users\Mark Hong\scoop\persist\vscode\data\extensions\iamhyc.overleaf-workshop-0.4.0\node_modules\ws\lib\WebSocket.js:442:14)
    at WebSocket.emit (node:events:513:28)
    at Receiver.ontext (C:\Users\Mark Hong\scoop\persist\vscode\data\extensions\iamhyc.overleaf-workshop-0.4.0\node_modules\ws\lib\WebSocket.js:841:10)
    at C:\Users\Mark Hong\scoop\persist\vscode\data\extensions\iamhyc.overleaf-workshop-0.4.0\node_modules\ws\lib\Receiver.js:536:18
    at Receiver.applyExtensions (C:\Users\Mark Hong\scoop\persist\vscode\data\extensions\iamhyc.overleaf-workshop-0.4.0\node_modules\ws\lib\Receiver.js:371:5)
    at C:\Users\Mark Hong\scoop\persist\vscode\data\extensions\iamhyc.overleaf-workshop-0.4.0\node_modules\ws\lib\Receiver.js:508:14
    at Receiver.flush (C:\Users\Mark Hong\scoop\persist\vscode\data\extensions\iamhyc.overleaf-workshop-0.4.0\node_modules\ws\lib\Receiver.js:347:3)
    at Receiver.finish (C:\Users\Mark Hong\scoop\persist\vscode\data\extensions\iamhyc.overleaf-workshop-0.4.0\node_modules\ws\lib\Receiver.js:541:12)
    at Receiver.expectHandler (C:\Users\Mark Hong\scoop\persist\vscode\data\extensions\iamhyc.overleaf-workshop-0.4.0\node_modules\ws\lib\Receiver.js:499:31)
    at Receiver.add (C:\Users\Mark Hong\scoop\persist\vscode\data\extensions\iamhyc.overleaf-workshop-0.4.0\node_modules\ws\lib\Receiver.js:103:24)
    at TLSSocket.realHandler (C:\Users\Mark Hong\scoop\persist\vscode\data\extensions\iamhyc.overleaf-workshop-0.4.0\node_modules\ws\lib\WebSocket.js:825:20)
    at TLSSocket.emit (node:events:513:28)
    at addChunk (node:internal/streams/readable:324:12)
    at readableAddChunk (node:internal/streams/readable:297:9)
    at TLSSocket.Readable.push (node:internal/streams/readable:234:10)
    at TLSWrap.onStreamRead (node:internal/stream_base_commons:190:23)
```

**Comment by iamhyc** [comment]
The connection problem is fixed in `v0.5.1`, please upgrade the extension to the latest version.

I will close this issue after 24 hours if there is no more problem.

**Comment by iamhyc** [comment]
> Thank you for developing a great feature. I am an Overleaf Premium user. I can log in Overleaf on VSCode, but I cannot get the project list as attached figure.
> 
> Is this error related to this issue?

Hi @taiga4112 , I think your problem is not related to the issue.

I have quickly check the code, the cause should be you have some project with multiple tags. This case is not handled correctly now.

https://github.com/iamhyc/Overleaf-Workshop/blob/693c4c180170f0ea5b35ed2ec85d9381b89c9464/src/core/projectManagerProvider.ts#L135

Let me know if you do have multiple tags on some project, and I will try to fix it then.

**Comment by taiga4112** [comment]
Hi, @iamhyc , thank you for your quick reply.
Yes, I have a lot of Overleaf projects with multiple tags as attached.

<img width="538" alt="スクリーンショット 2023-11-30 23 06 58" src="https://github.com/iamhyc/Overleaf-Workshop/assets/12507469/40af4e30-4004-41d6-a9d7-86597e4498b7">

**Comment by iamhyc** [comment]
> The connection problem is fixed in `v0.5.1`, please upgrade the extension to the latest version.
> 
> I will close this issue after 24 hours if there is no more problem.

Seems like nothing else happened. Will close it now.
